multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -2057,6 +2066,43 @@ def predict(
         if (as_pandas is None and isinstance(data, pd.DataFrame)) or as_pandas is True:
             pred = self._as_pandas(data=data, to_be_converted=pred)
 
+        if save_results:
+            if not result_path:
+                utcnow = datetime.utcnow()
+                timestamp = utcnow.strftime("%Y%m%d_%H%M%S")
+
+                result_path = f"Result{os.path.sep}ag-{timestamp}{os.path.sep}" \
+                              f"{self._hyperparameters['model.mmdet_image.checkpoint_name']}{os.path.sep}"
+                for i in range(1, 1000):
+                    try:
+                        os.makedirs(result_path, exist_ok=False)
+                        break
+                    except FileExistsError:
+                        logger.log(25, f"result_path {result_path} already exists")
+                else:
+                    raise RuntimeError("more than 1000 jobs launched in the same second")
+                logger.log(25, f'No result_path specified. Results will be saved in: "{result_path}"')
+                result_path = os.path.join(result_path, "results.ext")
+            ## Dumping Result for detection only now
+            if self._pipeline == OBJECT_DETECTION and detection_data_path:
+                if os.path.isdir(detection_data_path):
+                    ## this is for voc format
+                    result_name, _ = os.path.splitext(result_path)
+                    result_path = result_name + ".npy"
+                    np.save(result_path, pred)
+                    logger.info(25, f"Saved detection result to {result_path}")
+                    print(f"Saved detection result to {result_path}")
+                else:
+                    ## this is for coco format
+                    coco_dataset = COCODataset(detection_data_path)
+                    result_name, _ = os.path.splitext(result_path)
+                    result_path = result_name + ".json"
+                    coco_dataset.save_result(
+                        pred, from_coco_or_voc(detection_data_path, "test"), save_path=result_path
+                    )
+                    logger.info(25, f"Saved detection result to {result_path}")
+                    print(f"Saved detection result to {result_path}")
```

Comment:
this may need to be refactored to save_coco_result (or other names) and put in utils